Add option to stop highlighting History

Edru2 opened the pull request:
#### Brief overview of PR changes/additions
Adds an option in the settings menu to stop highlighting history.

#### Motivation for adding to Mudlet
several discussions in 
fix #4113 
fix #1682

#### Other info (issues closed, discussion etc)
If activated autocomplete as described in https://github.com/Mudlet/Mudlet/issues/4113#issuecomment-701540284 won't work (tab autocomplete still works as expected)

But as I already mentioned in that issue I think that is acceptable as people who activate this option didn't use/care about that way of autocomplete.

Edru2 committed:
add checkbox

Edru2 committed:
don't highligh history

Edru2 committed:
change checkbox name

vadi2 reviewed (approved):
The option addition seems fine, people want it - but I think it's more useful to it be "Highlight history" that's on by default, rather than the workaround way of calling it "Stop highlighting history"

Edru2 committed:
change unHighlight to Highlight history

option is checked by default,
even if no option in the xml is given as otherwhise it would be unchecked for all profiles created prior this version

vadi2 commented on src/XMLexport.cpp:
```diff
@@ -371,6 +371,7 @@ void XMLexport::writeHost(Host* pHost, pugi::xml_node mudletPackage)
     // that can be expressed solely with the Latin1 character encoding so that
     // can be used compared to the more complex Utf8 one needed otherwise:
     host.append_attribute("autoClearCommandLineAfterSend") = pHost->mAutoClearCommandLineAfterSend ? "yes" : "no";
+    host.append_attribute("HighlightHistory") = pHost->mHighlightHistory ? "yes" : "no";
```
This works well but double negatives are a bit tough to follow, how about using the `attributes().hasAttribute(` pattern from below

vadi2 commented on src/ui/profile_preferences.ui:
```diff
@@ -496,6 +496,16 @@
+             <string>&lt;p&gt;Hightlight History&lt;/p&gt;&lt;p&gt;&lt;i&gt;highlights your input line text if scrolling through your history&lt;/i&gt;&lt;/p&gt;</string>
+            </property>
+            <property name="text">
+             <string>Highlight History</string>
```
```suggestion
             <string>Highlight history</string>
```

The capitalisation in a label isn't needed

vadi2 commented on src/ui/profile_preferences.ui:
```diff
@@ -496,6 +496,16 @@
           <item row="1" column="1">
+           <widget class="QCheckBox" name="checkBox_highlightHistory">
+            <property name="toolTip">
+             <string>&lt;p&gt;Hightlight History&lt;/p&gt;&lt;p&gt;&lt;i&gt;highlights your input line text if scrolling through your history&lt;/i&gt;&lt;/p&gt;</string>
```
```suggestion
             <string>Highlights your input line text when scrolling through your history for easy cancellation</string>
```

Edru2 committed:
added suggestions
